lib/IDE/Refactoring.cpp

```diff
@@ -7357,21 +7450,65 @@ class AsyncConverter : private SourceEntityWalker {
       OS << tok::kw_try << " ";
     }
     OS << "await ";
-    addRange(CE->getStartLoc(), CE->getFn()->getEndLoc(),
-             /*ToEndOfToken=*/true);
+
+    // If there's no alternative then continue, but assume that the alternative
+    // has the same name as the callee
+    if (!HandlerDesc.Alternative) {
+      addRange(CE->getStartLoc(), CE->getFn()->getEndLoc(),
+               /*ToEndOfToken=*/true);
+    } else {
+      addRange(CE->getStartLoc(), CE->getFn()->getStartLoc(),
+               /*ToEndOfToken=*/false);
```

Comment:
Yes, not sure what I was thinking there.

In adding a test for that I found another bug where we weren't walking over the DeclRefExpr in a double curry thunk. Also since I just committed the getter fix I'll add a test and fix whatever that breaks as well :)